t/e2e/whisper/whisper_jail_test.go

```diff
@@ -180,16 +180,17 @@ func (s *WhisperJailTestSuite) TestJailWhisper() {
 
 				// generate symmetric key
 				var keyid = shh.newSymKey();
-				if (!shh.hasSymKey(keyid)) {
-					throw new Error('key not found');
-				}
 
 				// start watching for messages
 				var filter = shh.newMessageFilter({
 					topics: [topic],
 					symKeyID: keyid
 				});
 
+				if (!shh.hasSymKey(keyid)) {
```

Comment:
this fix looks very strange, what am i missing?

newSymKey inserts key and the whole operation is protected by rwmutex, hasSynKey is protected with rlock part of the same mutex. Given that all these operations are executed on the same thread, what was fixed with this commit?